### PR TITLE
DM-18122 update Spatial/temporal search panels

### PR DESCRIPTION
**Pull request by cwang2016**
This PR includes development for https://jira.lsstcorp.org/browse/DM-18122 
- add checkbox to the disclosure bar on the spatial/temporal search panels
- fix the size of the search panel while moving the resizer in the 'Select Constraints' area. 
- remove all-sky from spatial search method 
- validate the field entries depending on the entered value and the checkbox status of the search panel (empty value is allowed in case the search panel is not checked)
 - add feedback regarding field validation on the disclosure bar when the search panel is checked.
- merge the change regarding column fields in spatial/temporal search panel from dev. 

test: 
start  https://irsawebdev9.ipac.caltech.edu/dm-18122/firefly/firefly-dev.html
for styling,
    - check the size and layout of each panel in 4.Select Constraints area. Check how the size of each panel changes by sliding the resizer and resizing the entire TAP search panel 

for spatial search or temporal search, 
      -  enable spatial search or temporal search by checking the checkbox in the search panel
      - invalid red mark is seen by the field  which has invalid value or empty value (only when the search panel is checked for the search). 
     - an error message pop-up is shown in case there is some invalid entry in the enabled search 
panel while 'search' or 'populate and edit ADQL' is clicked. 
     - test example for spatial search,
         TAP service: IRSA, schema: wise_allwise, table: allwise_p3as_psd
         check Spatial search panel (and uncheck temporal search panel)
         target-> m31 , radius -> 200, and search or populate ADQL again
    - test case for temporal search,
        TAP service: CADC, schema: caom, table: caom2.plane
        check Temporal search panel (and uncheck spatial search panel)
        Temporal column: 'time_bounds_lower'
        time mode MJD mode
        enter '47600' and '47690' for 'from' and 'to' time.

### CODE REVIEW

**Comment by robyww** [comment]
I think you need to try to make the panel more narrow to give more room for the tables. I am attaching a diagram of my suggestions. Also I think you should add a little more vertical space between each of the disclosure boxes.

target-spacial

File: src/firefly/js/ui/tap/TableSearchMethods.jsx [hunk 1]
```diff
@@ -727,36 +803,65 @@ function tapSearchMethodReducer(columnsModel) {
...
+                const cols = getAvailableColumns(columnsModel);
+                set(rFields, [CrtColumnsModel, 'value'], columnsModel.tbl_id );
+                const centerColObj = formCenterColumns(columnsModel);
+                Object.assign(rFields[CenterLonColumns],
+                                       {validator: getColValidator(cols, false), value: centerColObj.lon, valid: true});
+                Object.assign(rFields[CenterLatColumns],
+                                       {validator: getColValidator(cols, false), value: centerColObj.lat, valid: true});
+
+                Object.assign(rFields[TemporalColumns],
+                                        {validator: getColValidator(cols, false), value: '', valid: true});
```
Comment (tgoldina) [inline src/firefly/js/ui/tap/TableSearchMethods.jsx]: The third argument to getColValidator should be false, if we don't want columns expressions. (The default is true.)

Comment (cwang2016) [reply src/firefly/js/ui/tap/TableSearchMethods.jsx]: updated. thanks.

File: src/firefly/js/ui/tap/TableSearchMethods.jsx [hunk 2]
```diff
@@ -49,52 +55,114 @@ const MJDTo = 'mjdto';
...
 const TimeTo = 'timeTo';
 const TimeOptions = 'timeOptions';
+const Wavelength = 'Wavelength';
+const WavelengthPanel = 'wavelengthSearchPanel';
+const WavelengthCheck = 'wavelengthCheck';
+const WavelengthColumns = 'wavelengthColumns';
+
+const CrtColumnsModel = 'crtColumnsModel';
+
+const SomeFieldInvalid = 'some field value is invalid';
```
[the remark this answers is not in the document]
Comment (cwang2016) [reply src/firefly/js/ui/tap/TableSearchMethods.jsx]: updated.

**Comment by cwang2016** [comment]
- the layout for the spatial/temporal is kept as original except adding more vertical space between the search panels. More update could be made once this PR is merged with the work made in https://jira.lsstcorp.org/browse/DM-18330 regarding the expansion of TAP search panel. 
- add parentheses around temporal search condition if after spatial search condition

**Pull request merged**